android/jni/app-android.cpp

```diff
@@ -1023,6 +1027,15 @@ extern "C" bool JNICALL Java_org_ppsspp_ppsspp_NativeActivity_runEGLRenderLoop(J
 
 	if (!graphicsContext->Init(wnd, desiredBackbufferSizeX, desiredBackbufferSizeY, backbuffer_format, androidVersion)) {
 		ELOG("Failed to initialize graphics context.");
+
+		if (vulkan && tries < 2) {
+			ILOG("Trying again, this time with OpenGL.");
+			g_Config.iGPUBackend = GPU_BACKEND_OPENGL;
+			SetGPUBackend((GPUBackend)g_Config.iGPUBackend);  // Wait, why do we need a separate enum here?
```

Comment:
Don't remember, maybe it was to avoid casting the class enum to an int or something?

-[Unknown]